healthz/healthz.proto

```diff
@@ -50,24 +65,134 @@ message GetRequest {
   types.Path path = 1;
 }
 
+message GetResponse {
+  ComponentStatus component = 1;
+}
+
 enum Status {
   STATUS_UNSPECIFIED = 0;   // The path doesn't support healthz
   STATUS_HEALTHLY = 1;      // The path is healthy
   STATUS_UNHEALTHY = 2;     // The path is unhealthy
 }
 
+message ArtifactHeader {
+  // ID of the artifact.
+  string id = 1;
+  // Artifact type describes data contents in the artifact.
+  // File artifacts should use the defined FileArtifactType.
+  // Proto artifacts should either use the generic ProtoArtifactType
+  // or the implementer can provide a specific artifact type
+  // which can add any additional metadata the implementor wants.
+  oneof artifact_type {
+    FileArtifactType file = 101;
+    ProtoArtifactType proto = 102;
+    google.protobuf.Any custom = 103;
+  }
+}
+
 message ComponentStatus {
   types.Path path = 1; // path of subcomponent.
+
   // Subcomponents that are aggregated by this status.
   repeated ComponentStatus subcomponents = 2;
+
   // Status of this component.
   Status status = 3;
+
   // Opaque data for how the healthcheck is implemented.  This can be any proto
   // defined by the vendor.  This could be the equivalent to outputs like
   // "show tech" or core files or any other diagnostic data.
-  google.protobuf.Any healthz = 4;
+  google.protobuf.Any healthz = 4 [deprecated=true];
+
+  // Artifacts provides links to all artifacts contained in this event.
+  // The individual artifacts can be retrieved via the Artifact() RPC.
+  repeated ArtifactHeader artifacts = 5;
+
+  // ID is the unique key for this event in the system.
+  string id = 6; 
+
+  // Acknowledged is set when a caller has processed the event.
+  bool acknowledged = 7;
+
+  // Created is the timestamp when this event was created.
+  google.protobuf.Timestamp created = 8;
+  
+  // Expires is the timestamp when the system will clean up the
+  // artifact. If unset, the artifact is not scheduled for GC.
+  google.protobuf.Timestamp expires = 9;
 }
 
-message GetResponse {
-  ComponentStatus component = 1;
+message ListRequest {
+  types.Path path = 1;
+  // By default only the unacknowledged events for the component will be
+  // returned.
+  bool include_acknowledged = 2;
+}
+
+message ListResponse {
+  repeated ComponentStatus status = 1;
+}
+
+message AcknowlegeRequest {
+  types.Path path = 1;
+  // Healthz event id.
+  string id = 2;
+}
+
+message AcknowledgeResponse {
+  ComponentStatus status = 1;
+}
+
+message ArtifactRequest {
+  // Artifact ID to be streamed.
+  string id = 1;
+}
+
+message ArtifactResponse {
+  oneof contents {
+    // Header is the first message in the stream. It contains
+    // the id of the artifact and metadata for the artifact
+    // based on the type of the artifact.
+    // OC defines FileArtifactType and ProtoArtifactType.
+    ArtifactHeader header = 1;
+    // Trailer contains the checksum for the streamed artifact.
+    Trailer trailer = 2;
```

Comment:
+1 to Brian's comment above, but also, should this be `ArtifactTrailer` if we keep it to ensure that this is consistently named?